tests/coverage/branch.coverage

```diff
@@ -0,0 +1,102 @@
+   LL|       |#![feature(coverage_attribute)]
+   LL|       |// edition: 2021
+   LL|       |// compile-flags: -Zunstable-options -Cinstrument-coverage=branch
+   LL|       |
+   LL|       |macro_rules! do_not_merge_with_signature {
+   LL|       |    () => { if true {} else {} };
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+   LL|       |}
+   LL|       |
+   LL|     15|fn if_not(cond: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|     15|    if !cond {
+  ------------------
+  |  Branch (LL:9): [True: 10, False: 5]
+  ------------------
+   LL|      5|        println!("cond was false");
+   LL|     10|    } else {
+   LL|     10|        println!("cond was true");
+   LL|     10|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_and(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|     15|    if a && b {
+                          ^12
+  ------------------
+  |  Branch (LL:8): [True: 12, False: 3]
+  |  Branch (LL:13): [True: 8, False: 4]
+  ------------------
+   LL|      8|        println!("both");
+   LL|      8|    } else {
+   LL|      7|        println!("not both");
+   LL|      7|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_or(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|     15|    if a || b {
+                          ^3
+  ------------------
+  |  Branch (LL:8): [True: 12, False: 3]
+  |  Branch (LL:13): [True: 14, False: 1]
+  ------------------
```

Comment:
Notice that the *true* count for `b` is wrong here, since it should be 2 instead of 14.